From b88b1b07e657716aa9975fee2dfc5a091c66a346 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 30 Jan 2017 11:35:15 +0000
Subject: [PATCH] Resolves: tdf#101664 dropcaps not set in styles on load

regression from...

commit 6766760e86b517c726204f9601b33a19cb0c5451
Date:   Thu Jan 14 21:05:59 2016 +0100

    refactor out to limit scope: RES_PARATR_DROP

    Change-Id: I8813b5676504e6de2af70d181fa9e905538fb05c

where MID_DROPCAP_CHAR_STYLE_NAME != nMemberId branch
in the past went to the default handler, while in
rework it ended up ignored

Change-Id: I8efa131750a064f594c1a2354d7aad19c2dc6b16
(cherry picked from commit d27aba549cdbcad6825ac6c955ee7d603aba1989)
Reviewed-on: https://gerrit.libreoffice.org/33696
Reviewed-by: Michael Stahl <mstahl@redhat.com>
Tested-by: Michael Stahl <mstahl@redhat.com>
---
 sw/source/core/unocore/unostyle.cxx | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sw/source/core/unocore/unostyle.cxx b/sw/source/core/unocore/unostyle.cxx
index e0d874f..7ec6ecb 100644
--- a/sw/source/core/unocore/unostyle.cxx
+++ b/sw/source/core/unocore/unostyle.cxx
@@ -1808,10 +1808,13 @@ void SwXStyle::SetPropertyValue<RES_TXTATR_CJK_RUBY>(const SfxItemPropertySimple
     SetPropertyValue<HINT_BEGIN>(rEntry, rPropSet, rValue, o_rStyleBase);
 }
 template<>
-void SwXStyle::SetPropertyValue<RES_PARATR_DROP>(const SfxItemPropertySimpleEntry& rEntry, const SfxItemPropertySet&, const uno::Any& rValue, SwStyleBase_Impl& o_rStyleBase)
+void SwXStyle::SetPropertyValue<RES_PARATR_DROP>(const SfxItemPropertySimpleEntry& rEntry, const SfxItemPropertySet& rPropSet, const uno::Any& rValue, SwStyleBase_Impl& o_rStyleBase)
 {
     if(MID_DROPCAP_CHAR_STYLE_NAME != rEntry.nMemberId)
+    {
+        SetPropertyValue<HINT_BEGIN>(rEntry, rPropSet, rValue, o_rStyleBase);
         return;
+    }
     if(!rValue.has<OUString>())
         throw lang::IllegalArgumentException();
     SfxItemSet& rStyleSet(o_rStyleBase.GetItemSet());
-- 
2.9.3

